source/extensions/filters/udp/udp_proxy/udp_proxy_filter.h

```diff
@@ -76,6 +111,12 @@ class UdpProxyFilter : public Network::UdpListenerReadFilter, Logger::Loggable<L
     UdpProxyFilter& parent_;
     const Network::UdpRecvData::LocalPeerAddresses addresses_;
     const Upstream::HostConstSharedPtr host_;
+    // TODO(mattklein123): Consider replacing an idle timer for each session with a last used
+    // time stamp and a periodic scan of all sessions to look for timeouts. This solution is simple,
```

Comment:
Might consider to use linked hash map to hold all the sessions in that case. 
It can also be one alarm per session, but scheduled once per loop instead of once per packet. Not necessarily to be a common idle timer cross all sessions.

Reply:
Agreed there is a bunch we can do. Let's tackle this in a follow up. Thanks for the good discussion. :)